internal/provider/warehouse_data_source.go

```diff
@@ -125,11 +130,9 @@ func (d *WarehouseDataSource) Read(ctx context.Context, req datasource.ReadReque
 			data.Region = types.StringNull()
 		}
 	}
-
-	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
-func getWarehouseByName(client tabular.Client, data WarehouseDataSourceModel, diag diag.Diagnostics) {
+func getWarehouseByName(client tabular.Client, data *WarehouseDataSourceModel, diag diag.Diagnostics) {
```

Comment:
This was doing a pass by copy before and the data was not being updated. I had to modify this in order to reuse the  functionality in the `ComputeConfigDataSource` and ensure original instance data was updated. Curious if was working here or users were always using the ID route and was just a hidden bug